source/core/src/main/com/csse3200/game/entities/Entity.java

```diff
@@ -36,6 +36,13 @@ public class Entity {
   private Vector2 position = Vector2.Zero.cpy();
   private Vector2 scale = new Vector2(1, 1);
   private Array<Component> createdComponents;
+  private EnemyType enemyType;
+  public enum EnemyType {
+    KANGAROO,
```

Comment:
Kangaroo is a boss entity no? I think it might be better to add a way to distinguish them from regular enemies? Just a thought